web/src/hotspots.ts

```diff
@@ -122,10 +135,10 @@ function place_icon(hotspot) {
     }
 
     const offset = {
-        top: $element.outerHeight() * hotspot.location.offset_y,
-        left: $element.outerWidth() * hotspot.location.offset_x,
+        top: ($element.outerHeight() ?? 0) * hotspot.location.offset_y,
+        left: ($element.outerWidth() ?? 0) * hotspot.location.offset_x,
```

Comment:
Similarly, probably `$element` could probably be redefined with a `.get(0)` included, and the checks above for whether `$element` is an actual thing reworked to result in the type system knowing `$element` has an `outerHeight` property.